javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -873,10 +878,23 @@ private Double position(TreeReference refAt) {
         return new Double(refAt.getMultLast());
     }
 
-    public static Object ifThenElse(DataInstance model, EvaluationContext ec, XPathExpression[] args, Object[] argVals) {

```

Comment:
Do you understand why this weird `argVals` param was being used? It's such an odd pattern that I'm afraid there was some obscure reason for it